src/ignored_suffixes.rs

```diff
@@ -0,0 +1,107 @@
+use std::ffi::OsStr;
+use std::fmt::Debug;
+use std::path::Path;
+
+use crate::error::*;
+
+#[derive(Debug, Clone)]
+pub struct IgnoredSuffixes<'a> {
+    values: Vec<&'a str>,
```

Comment:
Any particular reason you did not use `Vec<String>` here? That would result in simpler code. And I would be surprised if it had any noticeable effect on performance. (But performance can be hard to guess right on.)

It doesn't matter super much since this is all internal code that we can change later without affecting the public API. But it is always nice to try to get it right the first time 🙂

Reply:
So I went with `Vec<String>` at first, but after moving it to `SyntaxMapping` structure, and noticing that most of other parts use string slices I decided to change it. Also, since moving to `SyntaxMapping` I had to introduce lifetime parameter `'a`, it felt natural to use string slices with the same lifetime.

Reply:
I took a deeper look and agree now that the new code matches the old code as good as it can.